lib/services/cordova-project-plugins-service.ts

```diff
@@ -0,0 +1,682 @@
+					} else {
+						// Check if plugin is installed for current configuration.
+						let installedPlugin = this.getInstalledPluginByName(pluginName);
+						if(installedPlugin) {
+							this.$logger.info("Plugin '%s' is already installed", pluginName);
+							return;
+						}
+					}
+				}
+			}
+
+			let pluginToAdd = this.getPluginByName(pluginName);
+			if(pluginToAdd.type === pluginsDataLib.PluginType.MarketplacePlugin) {
+				version = this.selectPluginVersion(version, pluginToAdd).wait();
+				if(!this.isPluginSupported(pluginToAdd, this.$project.projectData.FrameworkVersion, version)) {
+					this.$errors.failWithoutHelp(`Plugin ${pluginName} is not avaialble for framework version '${this.$project.projectData.FrameworkVersion}'.`);
+				}
+			}
+
+			let configurations = this.$project.configurations;
+			if(_(this.pluginsForbiddenConfigurations).keys().find(key => key === pluginToAdd.data.Identifier)) {
+				let forbiddenConfig = this.pluginsForbiddenConfigurations[pluginToAdd.data.Identifier];
+				if(this.$project.configurations.length === 1 && _.contains(this.$project.configurations, forbiddenConfig)) {
+					this.$errors.failWithoutHelp(`You cannot enable plugin ${pluginName} in ${forbiddenConfig} configuration.`);
+				}
+				configurations = _.without(this.$project.configurations, forbiddenConfig);
+			}
+
+			this.configurePlugin(pluginName, version, configurations).wait();
+		}).future<void>()();
+	}
+
+	public removePlugin(pluginName: string): IFuture<void> {
+		return (() => {
+			if(!pluginName) {
+				this.$errors.fail("No plugin name specified.");
+			}
+
+			let installedPlugins = this.getInstalledPluginByName(pluginName);
+			let plugin = installedPlugins[0];
+			if(!plugin) {
+				this.$errors.fail("Could not find plugin with name %s.", pluginName);
+			}
+
+			if(this.$project.hasBuildConfigurations()) {
+				let configurations = this.$project.configurations;
+				_.each(configurations,(configuration: string) => {
+					this.removePluginCore(pluginName, plugin, configuration).wait();
+				});
+			} else {
+				this.removePluginCore(pluginName, plugin).wait();
+			}
+		}).future<void>()();
+	}
+
+	public printPlugins(plugins: IPlugin[]): void {
+		if(this.$options.available) {
+			// Group marketplace plugins
+			let marketplacePlugins = _.filter(plugins,(pl) => pl.type === pluginsDataLib.PluginType.MarketplacePlugin);
+			let output = _.filter(plugins, pl => pl.type === pluginsDataLib.PluginType.CorePlugin || pl.type === pluginsDataLib.PluginType.AdvancedPlugin);
+
+			let groups = _.groupBy(marketplacePlugins,(plugin: IPlugin) => plugin.data.Identifier);
+			_.each(groups,(group: any) => {
+				let defaultData = _.find(group,(gr: IPlugin) => {
+					let pvd = (<any>gr).pluginVersionsData;
+					return pvd && gr.data.Version === pvd.DefaultVersion;
+				});
+				if(defaultData) {
+					output.push(defaultData);
+				}
+			});
+
+			this.printPluginsCore(output);
+		} else {
+			this.printPluginsCore(plugins);
```

Comment:
The `if...else` statement leads to the same thing - a variable holding the end result can be introduced and `this.printPluginsCore` can be called at the end, omitting the `else`